website/docs/cdktf/examples-and-guides/refactoring.mdx

```diff
@@ -0,0 +1,255 @@
+refactoring-example  No changes. Your infrastructure matches the configuration.
+...
+```
+
+### Why do I need to find the terraform name of the resource?
+
+If you use Constructs to organize your code you might have a level between the generated provider constructs and the `TerraformStack` construct for code organization purposes.
+This extra level adds a prefix to the name of the resource to ensure uniqueness. Read more about this in the [Constructs](/cdktf/concepts/constructs#scope) section.
+
+## Moving resources from one stack to another
+
+Let's say you have a stack deployed with an aws `S3Bucket` resource and you want to move it to another stack:
+
+```typescript
+import { Construct } from "constructs";
+import { App, TerraformStack } from "cdktf";
+import { AwsProvider } from "@cdktf/provider-aws/lib/provider";
+import { S3Bucket } from "@cdktf/provider-aws/lib/s3-bucket";
+
+class MyFirstStack extends TerraformStack {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    new AwsProvider(this, "aws", {
+      region: "us-east-1",
+    });
+
+    // Old Definition
+    // new S3Bucket(this, "my-bucket", {
+    //   bucketPrefix: "my-bucket",
+    // });
+  }
+}
+
+class MySecondStack extends TerraformStack {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    new AwsProvider(this, "aws", {
+      region: "us-east-1",
+    });
+
+    // New Definition
+    new S3Bucket(this, "my-other-bucket", {
+      bucketPrefix: "my-bucket",
+    });
+  }
+}
+
+const app = new App();
+new MyFirstStack(app, "refactoring-example-stack-move-1");
+new MySecondStack(app, "refactoring-example-stack-move-2");
+app.synth();
+```
+
+When we run a `cdktf deploy '*'` we will see the following output:
+
+```shell-session
+...
+refactoring-example-stack-move-1    # aws_s3_bucket.my-bucket will be destroyed
+                                    # (because aws_s3_bucket.my-bucket is not in configuration)
+                                    - resource "aws_s3_bucket" "my-bucket" {
+...
+refactoring-example-stack-move-2    # aws_s3_bucket.my-other-bucket (my-other-bucket) will be created
+...
+```
+
+If we want to preserve the state of the resource we need to move it from one stack to another. For this we need to use [`terraform import`](/cli/commands/import) and [`terraform state rm`](/cli/commands/state/rm).
+
+```shell-session
+# First we need to find the terraform name of the resource in the old stack
+# Run this before removing the resource from the old stack
+$ cat cdktf.out/stacks/refactoring-example-stack-move-1/cdk.tf.json | jq '.resource.aws_s3_bucket'
+{
+  "my-bucket": {
+    "//": {
+      "metadata": {
+        "path": "refactoring-example-stack-move-1/my-bucket",
+        "uniqueId": "my-bucket"
+      }
+    },
+    "bucket_prefix": "my-bucket"
+  }
+}
+
+$ cdktf synth # To ensure our synthesized stack is up to date
+
+# Now we need the name of the resource in the new stack to understand
```

Comment:
Similar to above, this can be broken apart to multiple code snippets + text

Reply:
+1 to this suggestion :)